formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/Helpers.kt

```diff
@@ -22,8 +22,15 @@ internal enum class ProtoWireType(val typeId: Int) {
     ;
 
     companion object {
+        private val entryArray = Array(8) { typeId ->
+            ProtoWireType.entries.find { it.typeId == typeId } ?: INVALID
+        }
+
         fun from(typeId: Int): ProtoWireType {
-            return ProtoWireType.entries.find { it.typeId == typeId } ?: INVALID
+            if (typeId < 0 || typeId >= entryArray.size) {
+                return INVALID
+            }
```

Comment:
From an optimization perspective it may be valid to not check for `typeId<0` (and maybe have a larger array of `INVALID`s) to avoid the if check (instead relying on the array check that does this too).

Reply:
The array bounds check will "give" us `ArrayIndexOutOfBounds`, not the `ProtoWireType.INVALID` :)

Anyway, the implementation was implicitly based on the idea that the only call site extracts 3 least-significant bits from an int before passing it down this methods, so when this method is inlined the compiler can get rid of the `if`.